extensions/wikia/AdEngine/js/template/roadblock.js

```diff
@@ -0,0 +1,89 @@
+/*global define, require*/
+define('ext.wikia.adEngine.template.roadblock', [
+	'ext.wikia.adEngine.adContext',
+	'ext.wikia.adEngine.context.uapContext',
+	'ext.wikia.adEngine.provider.gpt.googleSlots',
+	'ext.wikia.adEngine.provider.gpt.helper',
+	require.optional('ext.wikia.adEngine.template.skin'),
+	'wikia.document',
+	'wikia.window',
+	'wikia.log'
+], function (
+	adContext,
+	uapContext,
+	googleSlots,
+	gptHelper,
+	skinTemplate,
+	doc,
+	win,
+	log
+) {
+	'use strict';
+
+	var logGroup = 'ext.wikia.adEngine.template.roadblock',
+		uapType = 'ruap',
+		medrecSlotEl = doc.getElementById('TOP_RIGHT_BOXAD');
+
+	/**
+	 * Handles medrec slot in case of UAP:roadblock
+	 * @param {Node} medrecSlotEl - medrec slot container element
+	 */
+	function handleMedrec(medrecSlotEl, uapId) {
+		var refreshed = false;
+
+		win.addEventListener('adengine.slot.status', function onAdStatusChange(event) {
```

Comment:
Maybe you could use `slotRegistry`?
https://github.com/Wikia/app/blob/d991351d3a4eff059f76eb69f3b6a8d98b8b8668/extensions/wikia/AdEngine/js/slot/service/slotRegistry.js#L30-L30

Reply:
It's very likely that you'll find targeting params there, so no need to parse HTML attributes.

Reply:
also you could use hooks: `slot.pre('success')`

Reply:
fixed